src/dft/backends/backend_compute_signature.cxx

```diff
@@ -0,0 +1,351 @@
+/*******************************************************************************
+* Copyright Codeplay Software Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+/*
+When only a specific backend library is required (eg. libonemkl_dft_<BACKEND>) 
+it may be preferenable to only link to that specific backend library without
+the requirement that the main OneMKL library also be linked.
```

Comment:
Could you please clarify this comment? Is it required for "compile time dispatching" case or there is another use case?

If I understand it correctly, introducing the same symbols (e.g., `oneapi::mkl::dft::compute_forward(...)`) in `libonemkl.so`, `libonemkl_dft_mklcpu.so`, and `libonemkl_dft_mklgpu.so` could cause symbol loading from incorrect library if all of these libraries will be loaded to the same application. How we can handle this problem?

Reply:
This addition addresses a [previous comment](https://github.com/oneapi-src/oneMKL/pull/259#discussion_r1083685852).

This is required for when dispatching to a specific backend library known at compile-time, with the idea that you'd only need to link against eg. `libonemkl_dft_mklgpu.so`. This avoids calls to `compute_forward(...)` having to go through the run-time dispatch mechanism.

If you were looking to use multiple DFT backends in the same application, the normal run-time dispatch mechanism must be used. This is because the intended backend used in `compute_forward(...)` must be found at run-time based on the `descriptor` argument. Because this mechanism loads the libraries at run-time, there is no issue caused by the same symbols. The user should not be linking against `libonekml_dft_XXXXXX.so`.

I can see that the symbol loading issue you describe might occur if a user wanted to use eg. BLAS functionality with a specific DFT backend. This would require linking against `libonemlk.so` and `libonemkl_dft_XXXXXX.so`. I think in those circumstances the first encountered `compute_forward` symbol would be used, and there wouldn't be an error. Calls to `compute_forward` would eventually call the intended function, either directly or via the function tables. I've not tested theory.

I think a challenge here is that the OneMKL spec does not state what should happen when the desired DFT backend is known at compile time. If we want to avoid the overhead of the runtime dispatch mechanism for `compute_forward/backward`, the only current way of doing this is by linking against a specific backend, and duplicating the symbols there. However, this adds the complications you describe.

Reply:
@mkrainiuk does the current solution work for you or support to link to libonemkl.so, libonemkl_dft_mklcpu.so and libonemkl_dft_mklgpu.so simultaneously is a requirement?